templates/edm.cpp.in

```diff
@@ -24,213 +24,40 @@
-
-      if (edmFieldIndices != indices)
-         continue;
-
-      const auto photonEmitterIndices = FermionVertex::template indices_of_field<2>(index);
-      const auto exchangeFieldIndices = FermionVertex::template indices_of_field<1>(index);
-      const auto vertex = FermionVertex::evaluate(index, context);
-
-      const auto photonEmitterMass = context.mass<PhotonEmitter>(photonEmitterIndices);
-      const auto exchangeFieldMass = context.mass<ExchangeField>(exchangeFieldIndices);
-
-      const double photonEmitterCharge =
-         - PhotonEmitter::electric_charge * unit_charge(context);
-
-      constexpr double numericFactor = oneOver16PiSquared;
-      const double massFactor = photonEmitterMass/(exchangeFieldMass * exchangeFieldMass);
-      const double couplingFactor = (std::conj(vertex.right()) * vertex.left()).imag();
-
-      const double massRatioSquared = Sqr(photonEmitterMass/exchangeFieldMass);
-      const double loopFactor = photonEmitterCharge * OneLoopFunctionA(massRatioSquared);
-      const double contribution = numericFactor * massFactor * couplingFactor * loopFactor;
-
-      res += contribution;
-   }
-
-   return res;
-}
-
-template<class EDMField, class PhotonEmitter, class ExchangeField>
-double EDMVertexCorrectionSF<
-EDMField, PhotonEmitter, ExchangeField
->::value(const typename field_indices<EDMField>::type& indices, context_base& context)
-{
-   double res = 0.0;
-
-   using FermionVertex = Vertex<
-                         EDMField,
-                         ExchangeField,
-                         PhotonEmitter
-                         >;
-
-   for (const auto& index: index_range<FermionVertex>()) {
-      const auto edmFieldIndices = FermionVertex::template indices_of_field<0>(index);
-
-      if (edmFieldIndices != indices)
-         continue;
-
-      const auto photonEmitterIndices = FermionVertex::template indices_of_field<2>(index);
-      const auto exchangeFieldIndices = FermionVertex::template indices_of_field<1>(index);
-      const auto vertex = FermionVertex::evaluate(index, context);
-
-      const auto photonEmitterMass = context.mass<PhotonEmitter>(photonEmitterIndices);
-      const auto exchangeFieldMass = context.mass<ExchangeField>(exchangeFieldIndices);
-
-      const double photonEmitterCharge =
-         - PhotonEmitter::electric_charge * unit_charge(context);
-
-      constexpr double numericFactor = oneOver16PiSquared;
-      const double massFactor = exchangeFieldMass/(photonEmitterMass * photonEmitterMass);
-      const double couplingFactor = (std::conj(vertex.right()) * vertex.left()).imag();
-
-      const double massRatioSquared = Sqr(exchangeFieldMass / photonEmitterMass);
-      const double loopFactor = photonEmitterCharge * OneLoopFunctionB(massRatioSquared);
-      const double contribution = numericFactor * massFactor * couplingFactor * loopFactor;
-
-      res += contribution;
-   }
-
-   return res;
 }
```

Comment:
Please add
~~~.cpp
// namespace CMSSMCPV_edm
~~~
as comment here

Reply:
Done in 80f97a15840a171faf0e764a0ebdfe6da432a8bc.